infrastructure/template.yaml

```diff
@@ -363,4 +364,127 @@ Resources:
               - kms:Decrypt
               - kms:GenerateDataKey*
               - kms:ReEncrypt*
-            Resource: !FindInMap [TxmaDummyInputQueue, !Ref Environment, KeyArn]
\ No newline at end of file
+            Resource: !FindInMap [TxmaDummyInputQueue, !Ref Environment, KeyArn]
+
+  ######################################
+  # OIDC UserInfo API Stub
+  ######################################
+  OidcUserInfoApiStubFunction:
+    Type: AWS::Serverless::Function
+    DependsOn:
+      - OidcUserInfoApiStubFunctionLogGroup
+    Properties:
+      FunctionName: !Sub ${Environment}-${AWS::StackName}-oidc-userinfo-stub
+      Architectures: ["arm64"]
+      CodeUri: ../oidc/userinfo/
+      Handler: oidc-userinfo-stub.handler
+      KmsKeyArn: !GetAtt LambdaKMSKey.Arn
+      PackageType: Zip
+      MemorySize: 128
+      ReservedConcurrentExecutions: 5
+      Runtime: nodejs18.x
+      Role: !GetAtt OidcUserInfoApiStubFunctionRole.Arn
+      Timeout: 5
+      Events:
+        userinfo:
+          Type: Api
+          Properties:
+            Path: /userinfo
+            Method: GET
+            RestApiId:
+              Ref: RestApi
+      VpcConfig:
+        SubnetIds:
+          - Fn::ImportValue: !Sub ${VpcStackName}-PrivateSubnetIdA
+          - Fn::ImportValue: !Sub ${VpcStackName}-PrivateSubnetIdB
+        SecurityGroupIds:
+          - Fn::ImportValue: !Sub ${VpcStackName}-AWSServicesEndpointSecurityGroupId
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - oidc-userinfo-stub.ts
+
+  OidcUserInfoApiStubFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      PermissionsBoundary: !If
+        - UsePermissionsBoundary
+        - !Ref PermissionsBoundary
+        - !Ref AWS::NoValue
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole
+        - !Ref OidcAuthorizeApiStubPolicy
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - "sts:AssumeRole"
+
+  OidcUserInfoApiStubFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Properties:
+      LogGroupName: !Sub "/aws/lambda/${Environment}-${AWS::StackName}-oidc-userinfo-api-stub"
+      RetentionInDays: 30
+      KmsKeyId: !GetAtt LoggingKmsKey.Arn
+
+  RestApi:
```

Comment:
is this the api for oidc stubs (account management api stub will have a seperate api) or for both?

Reply:
So I was thinking we can use the same for all that way the DNS can also be shared so Domain Name would be am-stub.home.${Environment}.account.gov.uk followed by the end points . But I think it also makes sense to keep the AM api and OIDC seperate ?

Reply:
yeah i think keeping some separation between AM and OIDC stubs would be good. Incase they need to go in different directions in future it will be a lot easier. And also a logical distinction